pkg/network/tracer/connection/kprobe/tracer.go

```diff
@@ -181,23 +167,41 @@ func LoadTracer(cfg *config.Config, mgrOpts manager.Options, perfHandlerTCP *dde
 
 	mgrOpts.ConstantEditors = append(mgrOpts.ConstantEditors, offsets...)
 
-	m, closeFn, err := prebuiltTracerLoader(cfg, mgrOpts, perfHandlerTCP, bpfTelemetry)
+	m, closeFn, err := prebuiltTracerLoader(cfg, mgrOpts, connCloseEventHandler, bpfTelemetry)
 	return m, closeFn, TracerTypePrebuilt, err
 }
 
-func loadTracerFromAsset(buf bytecode.AssetReader, runtimeTracer, coreTracer bool, config *config.Config, mgrOpts manager.Options, perfHandlerTCP *ddebpf.PerfHandler, bpfTelemetry *ebpftelemetry.EBPFTelemetry) (*manager.Manager, func(), error) {
+func loadTracerFromAsset(buf bytecode.AssetReader, runtimeTracer, coreTracer bool, config *config.Config, mgrOpts manager.Options, connCloseEventHandler ddebpf.EventHandler, bpfTelemetry *ebpftelemetry.EBPFTelemetry) (*manager.Manager, func(), error) {
 	m := ebpftelemetry.NewManager(&manager.Manager{}, bpfTelemetry)
-	if err := initManager(m, perfHandlerTCP, runtimeTracer, config); err != nil {
+
+	if err := initManager(m, connCloseEventHandler, runtimeTracer, config); err != nil {
 		return nil, nil, fmt.Errorf("could not initialize manager: %w", err)
 	}
 
+	ringbufferEnabled := util.RingBufferSupported(config)
+	util.AddBoolConst(&mgrOpts, ringbufferEnabled, "ringbuffers_enabled")
+	if ringbufferEnabled {
+		mgrOpts.MapSpecEditors[probes.ConnCloseEventMap] = manager.MapSpecEditor{
+			Type:       ebpf.RingBuf,
+			MaxEntries: uint32(util.ComputeDefaultClosedConnRingBufferSize()),
+			KeySize:    0,
+			ValueSize:  0,
+			EditorFlag: manager.EditType | manager.EditMaxEntries | manager.EditKeyValue,
+		}
+	}
+
```

Comment:
I see this code block repeated for fentry. Can we make a helper function to call from both places?

Reply:
done